.github/pipelines.sh

```diff
@@ -2,11 +2,18 @@
 set -e
 YQ="${YQ:-yq}"
 
+VERSION=$($YQ -V |rev|  cut -d " " -f 1 |rev| cut -d "." -f 1)
+if [[ "$VERSION" == "3" ]]; then
+    echo "yq version 3 detected, only version 4 is supported"
+    exit 1
+fi
+
 for i in $(ls config); do
     filename=$(basename -s .yaml $i)
-    for flavor in $(${YQ} r config/"$i" -p "p" 'flavors.[*]' | cut -d "." -f 2); do
-      for arch in $(${YQ} r config/"$i" -p "p" "flavors.$flavor.arches.[*]" | cut -d "." -f 4); do
-        ${YQ} r -X config/"$i" "flavors.$flavor.arches.$arch" | gomplate --left-delim "{{{" --right-delim "}}}" -V --datasource config="stdin:/?type=application/yaml" --file build.yaml.gomplate --out workflows/build-"$filename"-"$flavor"-"$arch".yaml
+    for flavor in $(${YQ} '.flavors|keys|join(" ")' config/"$i"); do
+      for arch in $(${YQ} ".flavors.$flavor.arches|keys|join(\" \")" config/"$i" ); do
```

Comment:
huh, much uglier!

Reply:
yes, I want to go afterwards and try to get it in a different way, Im pretty sure we can spit pairs of flavor+arch like we did on the img-builder script instead.

Although Im not sure which one is uglier lol, mixing bash and yaml ist not nice at all.